### DISCUSSION TITLE
Why is .env.local file not loaded in the test environment

### DISCUSSION THREAD

**Discussion by jmflannery** [bkeepers/dotenv#499] [Q&A]
I am curious why the `.env.local` file is not loaded in the test environment.

I run my test environment more often on my local machine than on CI. The development and test environments on my local machine share the same database configuration.

I want to share the same `.env.local` in my development and test environments. I prefer to check in the `.env` file as an example with no values. Instead of having one `.env.local`, I usually need a `.env.development.local` and `.env.test.local` with the same contents.

I understand this is the intended behavior. Is there any file that could be shared between development and test with higher precedence than `.env`?

Thank you!

Edit: I did add the [code as suggested](https://github.com/bkeepers/dotenv?tab=readme-ov-file#customizing-rails) into `application.rb` to add `.env.local` to the files to be loaded, and it is working as expected.

**Comment by bkeepers** [comment]
:wave: @jmflannery. See https://github.com/bkeepers/dotenv/issues/418 for history and discussion.
